src/change_trust_asset.js

```diff
@@ -0,0 +1,185 @@
+ * @param {LiquidityPoolParameters} liquidityPoolParameters – The liquidity pool parameters.
+ * @param {Asset} liquidityPoolParameters.assetA – The first asset in the Pool, it must respect the rule assetA < assetB. See validateLexicographicAssetsOrder() for more details on how assets are sorted.
+ * @param {Asset} liquidityPoolParameters.assetB – The second asset in the Pool, it must respect the rule assetA < assetB. See validateLexicographicAssetsOrder() for more details on how assets are sorted.
+ * @param {number} liquidityPoolParameters.fee – The liquidity pool fee. For now the only fee supported is `30`.
+ */
+export class ChangeTrustAsset extends Asset {
+  constructor(code, issuer, liquidityPoolParameters) {
+    if (!code && !issuer && !liquidityPoolParameters) {
+      throw new Error(
+        'Must provide either code, issuer or liquidityPoolParameters'
+      );
+    }
+
+    if ((code || issuer) && liquidityPoolParameters) {
+      throw new Error(
+        'Must provide either code (and optionally issuer) or liquidityPoolParameters but not both'
+      );
+    }
+
+    if (!liquidityPoolParameters) {
+      super(code, issuer);
+      this.xdrClass = xdr.ChangeTrustAsset;
+      return;
+    }
+
+    // We need to call the super constructor this way because Asset requires a valid code and issuer.
+    super('XLM');
+    this.code = undefined;
+
+    // Validate liquidity pool params.
+    const { assetA, assetB, fee } = liquidityPoolParameters;
+    if (!assetA || !(assetA instanceof Asset)) {
+      throw new Error('assetA is invalid');
+    }
+    if (!assetB || !(assetB instanceof Asset)) {
+      throw new Error('assetB is invalid');
+    }
+    if (!fee || fee !== LiquidityPoolFeeV18) {
+      throw new Error('fee is invalid');
+    }
+    this.liquidityPoolParameters = liquidityPoolParameters;
+    this.xdrClass = xdr.ChangeTrustAsset;
+  }
+
+  /**
+   * Returns a change trust asset object of the type "credit asset". Can be a
+   * native or an issued asset.
+   * @static
+   * @param {string} code - The asset code.
+   * @param {string} issuer - The account ID of the asset issuer.
+   * @return {ChangeTrustAsset}
+   * @memberof ChangeTrustAsset
+   */
+  static creditAsset(code, issuer) {
+    return new ChangeTrustAsset(code, issuer);
+  }
+
+  /**
+   * Returns a change trust asset object of the type "liquidity pool shares".
+   * @static
+   * @param {LiquidityPoolParameters} liquidityPoolParameters – The liquidity pool parameters.
+   * @param {Asset} liquidityPoolParameters.assetA – The first asset in the Pool, it must respect the rule assetA < assetB.
+   * @param {Asset} liquidityPoolParameters.assetB – The second asset in the Pool, it must respect the rule assetA < assetB.
+   * @param {number} liquidityPoolParameters.fee – The liquidity pool fee. For now the only fee supported is `30`.
+   * @return {ChangeTrustAsset}
+   * @memberof ChangeTrustAsset
+   */
+  static liquidityPoolAsset(liquidityPoolParameters) {
```

Comment:
WDYT of `liquidityPoolShare`?
```suggestion
  static liquidityPoolShare(liquidityPoolParameters) {
```

Reply:
Good suggestion. Done in https://github.com/stellar/js-stellar-base/pull/454/commits/185002ed3c232306b01057dd31d01d513574960c